zeroex/orderwatch/order_watcher.go

```diff
@@ -663,6 +453,401 @@ func (w *Watcher) handleBlockEvents(
 	return nil
 }
 
+// RevalidateOrdersForMissingEvents checks all of the orders in the database for
+// any events in the miniheaders table that may have been missed. This should only
+// be used on startup, as there is a different mechanism that serves this purpose
+// during normal operation.
+//
+// NOTE(jalextowle): This function can miss block events if the blockwatcher was
+// behind by more than db.MaxMiniHeaders when `handleBlockEvents` was last called.
+// This is extremely unlikely, so we have decided not to implement more costly
+// mechanisms to prevent from this possibility from occuring.
+func (w *Watcher) RevalidateOrdersForMissingEvents(ctx context.Context) error {
```

Comment:
Can you add a test for this method (either by calling it directly or setting up the database and then calling `Watcher.Watch`)?